src/content/docs/en/guides/environment-variables.mdx

```diff
@@ -93,22 +121,181 @@ const data = fetch(`${import.meta.env.PUBLIC_POKEAPI}/pokemon/squirtle`);
+      PORT: envField.number({ context: "server", access: "public", default: 4321 }),
+      API_SECRET: envField.string({ context: "server", access: "secret" }),
+    }
+  }
+})
+```
+
+Types will be generated for you when running `astro dev` or `astro build`, but you can run `astro sync` to generate types only.
+
+#### Use variables from your schema
+
+Import and use your defined variables from the appropriate `/client` or `/server` module:
+
+```astro
+---
+import { API_URL } from "astro:env/client"
+import { API_SECRET_TOKEN } from "astro:env/server"
+
+const data = await fetch(`${API_URL}/users`, {
+	method: "GET",
+	headers: {
+		"Content-Type": "application/json",
+		"Authorization": `Bearer ${API_SECRET_TOKEN}`
+	},
+})
+---
+
+<script>
+import { API_URL } from "astro:env/client"
+
+fetch(`${API_URL}/ping`)
+</script>
+```
+
+### Variable types
+
+There are three kinds of environment variables, determined by the combination of `context` (client or server) and `access` (secret or public) settings defined in your schema:
+
+- **Public client variables**: These variables end up in both your final client and server bundles, and can be accessed from both client and server through the `astro:env/client` module:
+
+   ```js
+   import { API_URL } from "astro:env/client"
+   ```
+
+- **Public server variables**: These variables end up in your final server bundle and can be accessed on the server through the `astro:env/server` module:
+
+   ```js
+   import { PORT } from "astro:env/server"
+   ```
+
+- **Secret server variables**: These variables are not part of your final bundle and can be accessed on the server through the `astro:env/server` module:
+
+   ```js
+   import { API_SECRET } from "astro:env/server"
+   ```
+
+   By default, secrets are only validated at runtime. You can enable validating private variables on start by [configuring `validateSecrets: true`](/en/reference/configuration-reference/#envvalidatesecrets).
+
+:::note
+**Secret client variables** are not supported because there is no safe way to send this data to the client. Therefore, it is not possible to configure both `context: "client"` and `access: "secret"` in your schema.
+:::
+
+### Data types
+
+There are currently four data types supported: strings, numbers, booleans and enums:
+
+```js
+import { envField } from "astro/config"
+
+envField.string({
+   // context & access
+   optional: true,
+   default: "foo",
+})
+
+envField.number({
+   // context & access
+   optional: true,
+   default: 15,
+})
+
+envField.boolean({
+   // context & access
+   optional: true,
+   default: true,
+})
+
+envField.enum({
+   // context & access
+   values: ['foo', 'bar', 'baz'],
+   optional: true,
+   default: 'baz',
+})
 ```
+
+Some of them come with more optional validation rules, check out the [source](https://github.com/withastro/astro/blob/main/packages/astro/src/env/schema.ts#L3-L45) to learn more.
```

Comment:
Is there a reason we are not *documenting* these? I would rather point people to a docs link than source code.

Reply:
Because we have no good place for it! It's not a runtime API thing, it's a configuration reference thing but that one is fully automated. IMO most people will discover these options with autocompletion in their editor

Reply:
Right, so the docs should be in config reference!

Reply:
I suggest we make a dummy `envField` entry in config reference locally in this branch, then go make one properly in the types/public file to fill it in.